src/btree/bt_sync_obsolete.c

```diff
@@ -290,3 +292,430 @@ __wt_sync_obsolete_cleanup(WT_SESSION_IMPL *session, WT_REF *parent)
 
     return (0);
 }
+
+/*
```

Comment:
Usually, internal threads are defined as `conn_*`, for instance, `conn_sweep`, `conn_compact`. Should we move this code to its own file?

Reply:
Thanks. I want to make the suggested changes as part of another PR once we decide the proper name for this thread.

Reply:
Sounds good!